### ISSUE TITLE
dbAppendTable not adding data to Hive table inside a schema

### ISSUE THREAD

**Issue by bjoerm** [r-dbi/DBI#250]
Hey guys,
Thanks a lot for all the work on connecting R with various databases!

I encountered problems when I tried to write data via DBI::dbAppendTable into a Hive database (that uses schemas).

I am using R 3.5, DBI 1.0.0, odbc 1.1.6 within RStudio 1.1.453 on a Windows 10 computer.

### Summary:
- Connecting to the database via odbc works fine.
- Executing dbGetQuery to SELECT, DROP, CREATE stuff works fine.
- dbCreateTable, dbRemoveTable, dbExistsTable and dbReadTable work fine when used in combination with DBI::Id.
- dbAppendTable however only returns a "0" in the R console (no error message) and does not add any values to tables.

### Details/Code (from our Hive database, so not easily reproduceabe):
Please find below the reduced code. Comments inside the code show the errors the were returned.

```r
# Packages ----------------------------------------------------------------

library(odbc)
library(DBI)
library(tibble)

# Connecting via odbc -----------------------------------------------------

con <- DBI::dbConnect(
  drv = odbc::odbc()
  , dsn = "hive" # Cloudera ODBC Driver for Apache Hive (64-bit)
)


# Setting up the path to the table
table_name <- DBI::Id(schema = "workspace_analytics", table = "bm_r_test")


# Trying to append data ---------------------------------------------------

# Following the example from https://www.r-dbi.org/blog/dbi-2-final/

DBI::dbCreateTable(con, table_name, c(a = "char")) # Works
DBI::dbAppendTable(con, table_name, data.frame(a = "tree"))
# Returns to console: [1] 0 and does not write values into the table.

DBI::dbReadTable(con, table_name) %>% tibble::as.tibble()
# Returns an empty tibble:
## A tibble: 0 x 1
## ... with 1 variable: a <chr>


DBI::dbRemoveTable(con, table_name)
DBI::dbCreateTable(con, table_name, c(a = "int", b = "float"))
DBI::dbAppendTable(con, table_name, data.frame(a = 1:3, b = 1:3))

DBI::dbReadTable(con, table_name) %>% tibble::as.tibble()
# Returns an empty tibble as well:
## A tibble: 0 x 2
## ... with 2 variables: a <int>, b <dbl>



# Approach via dbWriteTable
DBI::dbRemoveTable(con, table_name)

DBI::dbWriteTable(con, table_name, value = data.frame(a = 1:3, b = 1:3))
# Error: Can't unquote `workspace_analytics`.`bm_r_test`



DBI::dbCreateTable(con, table_name, c(a = "char")) # Works
DBI::dbWriteTable(con, table_name, data.frame(a = "tree"), append = TRUE)
# Error: Can't unquote `workspace_analytics`.`bm_r_test`



DBI::dbReadTable(con, table_name_mtcars) # Thus, returns a table with correct columns but with zero rows.
```

tl;dr: dbAppendTable does not append data into our Hive database. Instead of showing an error, it just returns a "0" to the console.

Can you help me out? Do you need any additional information? Do you have a lead on what I might try (e.g. how to get a closer look on what dbAppendTable is doing here and why it is not returning any error).

Again, thanks a lot for your work!

**Comment by bjoerm** [comment]
In addition to my description above, our data engineers provided me with the error message on the Hive server, which is 
```
2018-07-19 07:41:39,842 ERROR [HiveServer2-Handler-Pool: Thread-128820]: metadata.Hive (Hive.java:getTable(1187)) - Table workspace_analytics not found: default.workspace_analytics table not found
```

**It thus seems that DBI::Id does not yet work with DBI::dbAppendTable correctly.** Or that I just did something wrong.

**Comment by bjoerm** [comment]
I had the chance to also test this on another database.

**dbAppendTable also does not work for me on a Microsoft SQL 2016 database. Interestingly, dbWriteTable with append = TRUE works fine and appends values.**


```r 

# Packages ----------------------------------------------------------------
library(odbc)
library(DBI)

# Connection options ------------------------------------------------------

con <- DBI::dbConnect(
  drv = odbc::odbc()
  , dsn = "..."
  , uid = "..."
  , pwd = "..."
  , dbname = "..."
  )

# Some data ---------------------------------------------------------------

data <- cars

# Working on the database -------------------------------------------------

# Setting up the path to the table
table_name <- DBI::Id(schema = "ORC", table = "bm_r_test1")

# Drop Table if exists
if(DBI::dbExistsTable(con, table_name) == TRUE) {
  DBI::dbRemoveTable(con, table_name)
}

# Creating the table
DBI::dbWriteTable(con, table_name, data) # Works. Table with values in it is created.

# Appending via dbWriteTable
DBI::dbWriteTable(con, table_name, data, append = TRUE, overwrite = FALSE) # Appending via dbWriteTable works.


DBI::dbAppendTable(con, table_name, data) # Does not work. Does not append any values to the table but returns a a 0 to the console.

# Close the ODBC connection
DBI::dbDisconnect(con)
```

I hope this helps. If you need any further input or have questions, let me know.
